Description: fix compilation errors on Debian GNU/Hurd
 Apparently on hurd-i386 __GLIBC__ is defined, but the pthread implementation
 is special and cannot be used with constexpr. Hence exclude __gnu_hurd__.
Author: Florian Schlichting <fsfs@debian.org>

--- a/src/thread/PosixCond.hxx
+++ b/src/thread/PosixCond.hxx
@@ -43,7 +43,7 @@
 	pthread_cond_t cond;
 
 public:
-#ifdef __GLIBC__
+#if defined(__GLIBC__) && !defined(__gnu_hurd__)
 	/* optimized constexpr constructor for pthread implementations
 	   that support it */
 	constexpr PosixCond():cond(PTHREAD_COND_INITIALIZER) {}
--- a/src/thread/PosixMutex.hxx
+++ b/src/thread/PosixMutex.hxx
@@ -41,7 +41,7 @@
 	pthread_mutex_t mutex;
 
 public:
-#ifdef __GLIBC__
+#if defined(__GLIBC__) && !defined(__gnu_hurd__)
 	/* optimized constexpr constructor for pthread implementations
 	   that support it */
 	constexpr PosixMutex():mutex(PTHREAD_MUTEX_INITIALIZER) {}
